### ISSUE TITLE
mount host machine folder

### ISSUE THREAD

**Issue by mafintosh** [mist64/xhyve#45]
hey! great project!
is there anyway to mount a folder from the host machine inside the vm?

**Comment by colby-swandale** [comment]
you could use a nfs mount.

**Comment by gregory** [comment]
could you give more details @colby-swandale

**Comment by colby-swandale** [comment]
@gregory there are [literally thousands of posts on how to do this.](https://www.google.com/search?q=nfs+server)

**Comment by gregory** [comment]
oh sorry. i thought you meant a clean way to do it.... dont want to setup nfs server just for that :(
i used nfs at the very beginning when guest addition was not in boot2docker yet. thx though

**Comment by huytd** [comment]
yes, I'm looking for something like shared folder in VirtualBox with xhyve :D is there any way to do that?

**Comment by huytd** [comment]
OK, Thank to @colby-swandale 
Running NFS server is not a big deal, it's very simple with just 1 command line.
Here is the detailed note on how to mount host machine folder:

**Host:** Mac OS X 
**Client:** TinyCore

## Step 1: Enable network in xhyverun
If you are using `xhyverun.sh` to boot the TinyCore, just delete the comment character (`#`) before NET config
```
# NET="-s 2:0,virtio-net"
```
to
```
NET="-s 2:0,virtio-net"
```

Start the TinyCore VM with `sudo`:

```
sudo ./xhyverun.sh
```

## Step 2: Setup NFS Server on Host
- Create `/etc/exports` file
```
sudo touch /etc/exports
sudo vi /etc/exports
```
- Config the Exports file with the following syntax:
```
/Folder/on/host -mapall=501
```
Each folder in a line. The `mapall` parameter indicate the ID of user that have permission to access that host folder.
To find your user ID, type:
```
id
```
This is important unless you want to get `Connection refused` error when mounting from client.

For example, this is my `/etc/exports` file:
```
/Users/huy/Shared -mapall=501
```

## Step 3: Start and check NFS Server
- Start the NFS server with the following command:

```
sudo nfsd start
```

or restart it if it already started.

```
sudo nfsd restart
```

- Check the shared folder with the following command:
```
showmount -e
```
Your shared folder should displayed.

## Step 4: Mount the host folder on guest machine
In TinyCore guest, create a folder somewhere to access the shared folder, for example:
```
mkdir ~/shared
```

Mount the host folder with the following command:
```
sudo mount -o nolock <host-ip>:/Users/huy/Shared/ ~/shared
```

Done!

**Comment by xez** [comment]
It would be cool if bhyve/xhyve had an implementation of `VirtFS/virtio-9p` as a *much* more efficient way to support shared folders. [u9fs](https://bitbucket.org/plan9-from-bell-labs/u9fs/) might be a good starting point for a VirtIO implementation and seems to be license compatible.

**Comment by stv0g** [comment]
There's a fork which implements virtio-9p here: https://github.com/jceel/xhyve

I didn't managed to test it yet. Documentation is missing.
Can we merge this to this repo maybe?
I would offer to write missing docs..

**Comment by mauron85** [comment]
Probably no news for most of you, but you could use https://syncthing.net/. Using it with alpine linux and OSX. Very easy to setup if you don't mind syncing is not instant thing.

**Comment by xpepermint** [comment]
Status?
